File: location_service/LocationService.Infrastructure/Repository/Mapper/LocationMapper.cs
using LocationService.Core.Domain.Entity;
using LocationService.Infrastructure.Repository.Model;

namespace LocationService.Infrastructure.Repository.Mapper
{
    public class LocationMapper
    {
        public static LocationModel ToModel(LocationEntity entity)
        {
            if (entity == null) return null!;
            return new LocationModel
            {
                Id = entity.Id,
                CountryId = entity.CountryId,
                ProvinceId = entity.ProvinceId,
                Latitude = entity.Latitude,
                Longitude = entity.Longitude,
                Detail = entity.Detail,
                Name = entity.Name
            };
        }

        public static LocationEntity ToEntity(LocationModel model)
        {
            if (model == null) return null!;
            return new LocationEntity
            {
                Id = model.Id,
                CountryId = model.CountryId,
                ProvinceId = model.ProvinceId,
                Latitude = model.Latitude,
                Longitude = model.Longitude,
                Detail = model.Detail,
                Name = model.Name
            };
        }
    }
}

File: accommodation_service/src/main/java/huy/project/accommodation_service/infrastructure/repository/mapper/LocationMapper.java
package huy.project.accommodation_service.infrastructure.repository.mapper;

import huy.project.accommodation_service.core.domain.entity.LocationEntity;
import huy.project.accommodation_service.infrastructure.repository.model.LocationModel;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

import java.util.List;

@Mapper
public abstract class LocationMapper {
    public static final LocationMapper INSTANCE = Mappers.getMapper(LocationMapper.class);

    public abstract LocationEntity toEntity(LocationModel model);
    public abstract LocationModel toModel(LocationEntity entity);

    public abstract List<LocationEntity> toListEntity(List<LocationModel> models);
}
